proof/refine/ARM/Reply_R.thy

```diff
@@ -953,34 +946,35 @@ lemma bindScReply_sch_act_wf[wp]:
   by (wpsimp wp: sts_sch_act' hoare_vcg_all_lift hoare_vcg_if_lift hoare_drop_imps)
 
 lemma bindScReply_sym_refs_list_refs_of_replies':
-  "\<lbrace>\<lambda>s. sym_refs (list_refs_of_replies' s) \<and>
-        obj_at' (\<lambda>ko. replyPrev ko = None \<and> replyNext ko = None) replyPtr s \<and>
-        (\<forall>oldReplyPtr. obj_at' (\<lambda>ko. scReply ko = Some oldReplyPtr) scPtr s \<longrightarrow>
-         obj_at' (\<lambda>ko. replyNext ko = Some (Head scPtr)) oldReplyPtr s)\<rbrace>
+  "\<lbrace>\<lambda>s. sym_refs (list_refs_of_replies' s)
+        \<and> \<not> is_reply_linked replyPtr s \<and> replySCs_of s replyPtr = None
+        \<and> (\<forall>oldReplyPtr. obj_at' (\<lambda>ko. scReply ko = Some oldReplyPtr) scPtr s
```

Comment:
I guess this `obj_at’` is needed. Can we change this to `sc_at’` and a projection, too?